### ISSUE TITLE
Weight Learning with Arithmetic Negative Prior Crashes

### ISSUE THREAD

**Issue by eriq-augustine** [eriq-augustine/psl#109]
To reproduce, just take any psl-example (easy lp cli in this example), and change the prior from logical:
```
5:  !Knows(P1,P2) ^2
```
to arithmetic:
```
5:  Knows(P1,P2) = 0.0 ^2
```

We get an exception like this:
```
Unexpected exception!
java.lang.NullPointerException
	at org.linqs.psl.reasoner.admm.ADMMReasoner.changedGroundKernelWeight(ADMMReasoner.java:228)
	at org.linqs.psl.reasoner.admm.ADMMReasoner.changedGroundKernelWeights(ADMMReasoner.java:239)
	at org.linqs.psl.application.learning.weight.maxlikelihood.VotedPerceptron.doLearn(VotedPerceptron.java:287)
	at org.linqs.psl.application.learning.weight.WeightLearningApplication.learn(WeightLearningApplication.java:120)
	at org.linqs.psl.cli.Launcher.run(Launcher.java:241)
	at org.linqs.psl.cli.Launcher.main(Launcher.java:351)
```

**Comment by eriq-augustine** [comment]
This may have been fixed with the changes in `lazy-grounding`.
Make sure that you check on that branch, or wait for the merge into `develop`.

**Comment by eriq-augustine** [comment]
I don't think this was fixed yet.
It looks much harder than at first glance.
I think that the parser is parsing the atom as a SummationAtom instead of a normal QueryAtom.
We will have to go back to the parser.
(It also works on some machines.)

**Comment by eriq-augustine** [comment]
Fixed in f8a8446fb71e82c3383bbed1ead0e6b533da6aec.
